src/app/profile/receipts/receipts.component.js

```diff
@@ -90,7 +90,7 @@ class ReceiptsController {
     );
   }
 
-  setYear(year) {
+  onYearChange(year) {
     this.maxShow = this.step;
     this.currentYear = year;
     this.getReceipts(year);
```

Comment:
Setting the year is handled by `ng-model`. So this method just needs to reload the receipts.

```suggestion
  onYearChange() {
    this.maxShow = this.step;
    this.getReceipts(this.currentYear);
```